lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -1324,6 +1324,19 @@ bool WitnessChecker::findBestWitness(
     anyFromUnconstrainedExtension = false;
     bestIdx = 0;
 
+    auto &ctx = getASTContext();
+    bool isEquatableConformance =
+        conformance && conformance->getProtocol() ==
+                           ctx.getProtocol(KnownProtocolKind::Equatable);
+
+    bool isSwiftRawRepresentableEnum = false;
+    if (conformance && conformance->getType()) {
+      auto enumType = conformance->getType()->getAs<EnumType>();
```

Comment:
This won't work if the enum is generic. You should use conformance->getDeclContext()->getSelfNominalDecl() instead